dask/dataframe/io/parquet.py

```diff
@@ -121,7 +117,8 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
                         columns=[c for c in pf.columns if c in dtypes])
 
     for cat in categories:
```

Comment:
Looks like you still need to handle the case if `categories` is `None` here. Also, what happens if the data is written as categories, but the user doesn't pass in any categories to `read_parquet`? Or if they pass in categories for a column that isn't?